pingcap/dm
# *: add an upgrade mechanism and fix task meta encoding

### What problem does this PR solve? 

restart DM-worker can be failed when having more than 256 task operations

### What is changed and how it works?

- add an upgrade mechanism
- change encoding for task meta from `LittleEndian` to `BigEndian`
	- when upgrading from an older version, clear previous saved meta

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
	 - use an older version of DM-worker to start some tasks
	 - stop DM-worker
	 - replace DM-worker build from this PR
	 - observe whether DM-worker started, task meta cleared and no task auto running

Code changes

 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
